lib/tools/cfgupgrade.py

```diff
@@ -762,11 +771,31 @@ def DowngradeXenSettings(self):
       if variant in hvparams:
         hvparams[variant]["xen_cmd"] = "xl"
 
+  @OrFail("Revert KVM parameter")
+  def DowngradeKVMSettings(self):
+    """Downgrades KVM Settings
+
+    """
+    # pylint: disable=E1103
+    # Because config_data is a dictionary which has the get method.
+    cluster = self.config_data.get("cluster", None)
+    if cluster is None:
+      raise Error("Can't find the cluster entry in the configuration")
+
+    hvparams = cluster.get("hvparams", None)
+    if hvparams is None:
+      return
+
+    if DOWNGRADE_MAJOR == 3 and DOWNGRADE_MINOR < 1:
+      if "debug_threads" in hvparams[constants.HT_KVM]:
+        del hvparams[constants.HT_KVM]["debug_threads"]
+
```

Comment:
`debug_threads` can also be set individually at instance level, right? Shouldn't the downgrade code also check and remove on every instance?

Reply:
Yes you are right, I guess it was a bit to late :-) I'll test for this scenario and add code if required.